### ISSUE TITLE
Unexpected config error when a default config file is empty

### ISSUE THREAD

**Issue by victor-yon** [omni-us/jsonargparse#367]
## 🐛 Bug report

If at least one `default_config_files` is set and this file is empty, a `TypeError: Unexpected config: ` is triggered on `parser.parse_args()`. Even if no argument is mandatory.

### To reproduce

```python
from jsonargparse import ArgumentParser

parser = ArgumentParser(default_config_files=['config.yaml'])
parser.add_argument("test", type=int, help="test", default=1)
cfg = parser.parse_args()
print(cfg.as_dict())
```

**no 'config.yaml' file**
```bash
python main.py
> {'test': 1}
```
Work as expected, no error

**empty 'config.yaml' file**
```bash
touch config.yaml
python main.py
> Unexpected config error:
```

**Not empty 'config.yaml' file**
Work as expected, no error
```bash
echo "test: 42" > config.yaml
python bug.py
> {'test': 42, '__default_config__': Path_fr(config.yaml, cwd=/Users/...)}
```

### Expected behavior

I expect to have the same behavior between no configuration file and empty configuration file.
In the case where every argument are optional I don't expect to have an error.

In my case I wanted to generate this default configuration file as suggest [in the document](https://jsonargparse.readthedocs.io/en/v4.24.0/#writing-configuration-files) with:

```bash
python example.py --print_config > config.yaml
````

But an empty file is created before to run the script. Which trigger the issue mentioned above.

### Environment

- jsonargparse version: 4.24.0
- Python version: 3.11.4
- How jsonargparse was installed: `pip install jsonargparse[signatures]~=4.24 docstring-parser~=0.15 PyYAML~=6.0 ruyaml~=0.91`
- OS: macOS 13.4.1

**Comment by mauvilsa** [comment]
@victor-yon thank you for reporting!

You are right. The recommended steps to create a config should not fail just because `default_config_files` is used. This should be fixed.

**Comment by mauvilsa** [comment]
@victor-yon I created pull request #370 fixing this. Please try it out.

**Comment by victor-yon** [comment]
Hi, thank you for the fix. I can try next Monday, I don't have access to my computer before.

**Comment by mauvilsa** [comment]
Merged it since I wanted to create a release and this change was simple. You can test it next week with the released package. If there are issues, please comment here.

**Comment by victor-yon** [comment]
I confirm that is working as expected with `jsonargparse v4.24.1`. Thank you.
